codersdk/workspaceagents.go

```diff
@@ -392,6 +392,68 @@ func (c *Client) WorkspaceAgentListeningPorts(ctx context.Context, agentID uuid.
 	return listeningPorts, json.NewDecoder(res.Body).Decode(&listeningPorts)
 }
 
+// WorkspaceAgentContainer describes a container of some sort that is visible
+// to the workspace agent.
+type WorkspaceAgentContainer struct {
+	// CreatedAt is the time the container was created.
+	CreatedAt time.Time `json:"created_at" format:"date-time"`
+	// ID is the unique identifier of the container.
+	ID string `json:"id"`
+	// FriendlyName is the human-readable name of the container.
+	FriendlyName string `json:"name"`
+	// Image is the name of the container image.
+	Image string `json:"image"`
+	// Labels is a map of key-value pairs of container labels.
+	Labels map[string]string `json:"labels"`
+	// Running is true if the container is currently running.
+	Running bool `json:"running"`
+	// Ports includes ports exposed by the container.
+	Ports []WorkspaceAgentListeningPort `json:"ports"`
+	// Status is the current status of the container. This is somewhat
+	// implementation-dependent, but should generally be a human-readable
+	// string.
+	Status string `json:"status"`
+	// Volumes is a map of "things" mounted into the container. Again, this
+	// is somewhat implementation-dependent.
+	Volumes map[string]string `json:"volumes"`
+}
+
+// WorkspaceAgentListContainersResponse is the response to the list containers
+// request.
+type WorkspaceAgentListContainersResponse struct {
+	Containers []WorkspaceAgentContainer `json:"containers"`
+}
+
+// WorkspaceAgentContainersLabelFilter is a RequestOption for filtering
+// listing containers by labels.
+func WorkspaceAgentContainersLabelFilter(kvs map[string]string) RequestOption {
```

Comment:
This seems like it could be a private function? Let's not cloud the API package with functions that aren't useful for the consumer.